src/sentry/api/serializers/models/group.py

```diff
@@ -405,12 +405,24 @@ def _get_permalink(self, obj, user):
                 request.auth, obj.organization
             )
 
-        if (
-            is_superuser
-            or (user.is_authenticated() and user.get_orgs().filter(id=obj.organization.id).exists())
-            or is_valid_sentryapp
-        ):
-            return obj.get_absolute_url()
+        with sentry_sdk.start_span(op="GroupSerializerBase.serialize.permalink.check") as span:
+
+            def user_is_in_org():
+                exists = user.get_orgs().filter(id=obj.organization.id).exists()
+                span.set_data("user_is_in_org", exists)
+                return exists
```

Comment:
The structure may look very different but this should actually be a no-op change.

This is a little baroque but I wanted to max out the amount of data we capture, at least temporarily. I wouldn't mind refactoring this back to its simpler form, once the spans are no longer useful.

Reply:
I figured as much.  I did have to look at it for a little bit to figure out what you were trying to do there.